platforms/core-configuration/model-core/src/main/java/org/gradle/internal/snapshot/impl/DefaultValueSnapshotter.java

```diff
@@ -157,6 +157,11 @@ public ValueSnapshot array(ImmutableList<ValueSnapshot> elements, Class<?> array
             return new ArrayValueSnapshot(elements);
         }
 
+        @Override
+        public ValueSnapshot primitiveArray(Object value) {
+            return javaSerialized(value, AbstractValueProcessor.javaSerialized(value));
```

Comment:
❓ Could there be any performance impact of using java serialization or should we implement something custom here?

Reply:
There could be. I thought that could be done as a separate piece of work.